honeybee/aperture.py

```diff
@@ -0,0 +1,506 @@
+# coding: utf-8
+"""Honeybee Aperture."""
+from .properties import ApertureProperties
+from .aperturetype import aperture_types, Window
+from .boundarycondition import Outdoors, Surface
+from .shade import Shade
+from .typing import valid_string
+import honeybee.writer as writer
+
+from ladybug_geometry.geometry3d.pointvector import Point3D, Vector3D
+from ladybug_geometry.geometry3d.face import Face3D
+
+import math
+
+
+class Aperture(object):
+    """A single planar aperture face.
+
+    Properties:
+        name
+        name_original
+        geometry
+        vertices
+        upper_left_vertices
+        triangulated_mesh3d
+        normal
+        center
+        area
+        perimeter
+        boundary_condition
+        parent
+        has_parent
+    """
+    TYPES = aperture_types
+    __slots__ = ('_name', '_name_original', '_geometry', '_parent',
+                 '_type', '_boundary_condition', '_properties', '_writer')
+
+    def __init__(self, name, geometry, type=None, boundary_condition=None):
+        """A single planar aperture in a face.
+
+        Args:
+            name: Aperture name. Must be < 100 characters.
+            geometry: A ladybug-geometry Face3D.
+            type: Aperture type. Default: Window
+            boundary_condition: Boundary condition object (Outdoors, Surface).
+                Default: Outdoors.
+        """
+        # process the name
+        self._name = valid_string(name, 'honeybee aperture name')
+        self._name_original = name
+
+        # process the geometry
+        assert isinstance(geometry, Face3D), \
+            'Expected ladybug_geometry Face3D. Got {}'.format(type(geometry))
+        self._geometry = geometry
+        self._parent = None  # _parent will be set when the Face is added to a Face
+
+        # process the type and boundary condition
+        self.type = type or Window()
+        self.boundary_condition = boundary_condition or Outdoors()
+
+        # initialize properties and writer for extensions
+        self._properties = ApertureProperties(self)
+        self._writer = writer
```

Comment:
I didn't really touch this part from your original code since it seems that you have a clearer idea about it and it's not something that I really plan on using much with Energy.  The fact that the OpenStudio pathway offers a lot more functionality and that all of these IDF methods will be deprecated by the time our SBIR is finished doesn't make a very compelling case to develop them very robustly or to support them on the plugin level. I will make this change here but I feel that it may be best if you develop out these writers for Radiance first and then I will follow your lead if we want to have a minimal version for energy.